tqdm/std.py

```diff
@@ -1442,27 +1474,10 @@ def self_wrapattr(self, stream, method, close=True):
         ...         if not chunk:
         ...             break
         """
-        func = getattr(stream, method)
-        if method == "write":
-            @wraps(func)
-            def wrapped(data, *args, **kwargs):
-                res = func(data, *args, **kwargs)
-                self.update(len(data))
-                return res
-        elif method == "read":
-            @wraps(func)
-            def wrapped(*args, **kwargs):
-                res = func(*args, **kwargs)
-                self.update(len(res))
-                return res
-        else:
-            raise TqdmKeyError("Can only wrap read/write methods")
-
         try:
-            setattr(stream, method, wrapped)
-            yield stream
+            wrapped = CallbackIOWrapper(self.update, stream, method)
+            yield wrapped
```

Comment:
You can move the context manager functionality into `CallbackIOWrapper` and you won't need two methods here.
```python
def __enter__(self):
    return self
```

Reply:
given this some more thought - I think you're right; now that I've abstracted most of the logic to `CallbackIOWrapper` the instance method convenience function is not required (it takes the same number of lines of code to use). This also gets rid of the annoying `close=True`, name clash, and generally seems cleaner.